data-prepper-plugins/s3-sink/src/main/java/org/opensearch/dataprepper/plugins/sink/s3/S3SinkService.java

```diff
@@ -135,9 +138,20 @@ void output(Collection<Record<Event>> records) {
                     failedEvents.add(event);
                 }
 
-                flushToS3IfNeeded();
+                final boolean flushed = flushToS3IfNeeded(currentBuffer);
+
+                if (flushed) {
+                    s3GroupManager.removeGroup(s3GroupEntry.getKey());
+                }
+            }
+
+            for (final Map.Entry<S3GroupIdentifier, S3Group> s3GroupEntry : s3GroupManager.getS3GroupEntries()) {
```

Comment:
should this be flushed when global threshold is reached ?

Reply:
The global threshold will be handled in a follow-up PR, and will be configured separately as `aggregate_threshold`

Reply:
I believe that @graytaylor0 will submit a follow-on to support the `aggregate_threshold` to do just that.